### ISSUE TITLE
OOM

### ISSUE THREAD

**Issue by yanqiangmiffy** [pranavajitnair/BiDAF#6]
OOM problems always occur, I would like to ask if it is because there is no Batch processing

**Comment by pranavajitnair** [comment]
It is probably because I have not used batching. I think reducing the number of iterations per epoch might solve the problem. Also
using torch.no_grad in the validation loop might solve the problem if OOM is happening in the validation loop.

**Comment by yanqiangmiffy** [comment]
Thank you for your prompt reply. OOM happened in the train step. I have tried using torch.no_grad and added it to line 20 of train.py, but loss.backward() will throw an exception, so I don’t really decide whether to use it in the training phase. torch.no_grad. Currently I try to reduce iterations

**Comment by pranavajitnair** [comment]
Did reducing the number of iterations per epoch work ?

**Comment by yanqiangmiffy** [comment]
yes thanks，i try to implement the bath datalaoder，So that all training sets can be used
